src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -1184,17 +1184,26 @@ static int client_parse_message(
 
                 case SD_DHCP6_OPTION_STATUS_CODE: {
                         _cleanup_free_ char *msg = NULL;
+                        bool ignore;
 
                         r = dhcp6_option_parse_status(optval, optlen, &msg);
                         if (r < 0)
                                 return r;
+                        if (r == 0)
+                                break;
+
+                        /* Those two errors are handled later in this function. */
+                        ignore = IN_SET(r, DHCP6_STATUS_NO_ADDRS_AVAIL, DHCP6_STATUS_NO_PREFIX_AVAIL);
 
-                        if (r > 0)
-                                return log_dhcp6_client_errno(client, SYNTHETIC_ERRNO(EINVAL),
-                                                              "Received %s message with non-zero status: %s%s%s",
-                                                              dhcp6_message_type_to_string(message->type),
-                                                              strempty(msg), isempty(msg) ? "" : ": ",
-                                                              dhcp6_message_status_to_string(r));
+                        log_dhcp6_client(client,
+                                         "Received %s message with non-zero status%s: %s%s%s",
+                                         dhcp6_message_type_to_string(message->type),
+                                         ignore ? ", ignoring" : "",
+                                         strempty(msg), isempty(msg) ? "" : ": ",
+                                         dhcp6_message_status_to_string(r));
+
+                        if (!ignore)
```

Comment:
Hm, will the PR result only in display of the other log message:

`return log_dhcp6_client_errno(client, SYNTHETIC_ERRNO(EINVAL), "No IA_PD prefix or IA_NA address received. Ignoring.");`

at string 1323/1332, because the client sends still a solicit message with the both `IA_NA` and `IA_PD` option and a server will respond `NoAddrsAvail` to it?

Reply:
The line should not be triggered if at least one prefix (or address) is provided. But, the message sent by your server does not include other information like DNS servers if a non-zero status code is included. I will also address that.